service/routes.py

```diff
@@ -77,7 +77,34 @@ def read_order(order_id):
 
     return jsonify(order.serialize()), status.HTTP_200_OK
 
+  
+######################################################################
+    # UPDATE AN ORDER
+######################################################################
+
+@app.route("/orders/<int:order_id>", methods=["PUT"])
+def update_order(order_id):
+    """Updates an order"""
+    app.logger.info(f"Request to update order id:{order_id}")
+
+    #Check if order exists
+    order = Order.find(order_id)
+    if not order:
+        abort(
+            status.HTTP_404_NOT_FOUND, f"Order with id '{order_id}' was not found."
+        )
+
+    # Update order with info in the json request
+    app.logger.debug("Payload = %s", api.payload)
```

Comment:
I ran this code and got: "api" is not defined. I guess you are extracting the json data here, you may use "request.get_json()".